### PR TITLE
add android import to xenial branch

### PR DESCRIPTION
**Pull request by delijati**
Android import sms, call history + README see #7

### CODE REVIEW

File: tools/backup/main.cpp [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            if (type == "2") {
+                senderid = "self";
+            }
+
+            map["time"]     = time;
+            map["senderid"] = senderid;
+            map["number"]   = number;
+            map["message"]  = message;
+
+            // qDebug() << "Entry: " << map;
```
Comment (UniversalSuperBox) [inline tools/backup/main.cpp]: We shouldn't include commented-out code sections in the codebase

File: tools/backup/calllogs_export.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (UniversalSuperBox) [inline tools/backup/calllogs_export.xml]: Should these files be called `_example.xml`? So people know that they need to put their own files here?

File: tools/backup/main.cpp [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    qDebug() << "Writing calls into db ...";
+    writeCalls(accountId, calls);
+
+    auto messages =
+        readMessages(QString(message_export));
+
+    qDebug() << "Extracted:" << messages.count() << "messages";
+    qDebug() << "Writing messages into db ...";
+    writeMessages(accountId, messages);
+    qDebug() << "Finished!";
```
Comment (UniversalSuperBox) [inline tools/backup/main.cpp]: You may wish to use `qInfo()` rather than `qDebug()`.

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]
Excellent, thank you! This will be awesome to have, and it's the basis for big things.

A few questions:
* If we separate the code out into a module, it is possible that we could create an "import" option in the interface of messaging-app and dialer-app. Would you be willing to take this on? Now or later?
* Shall we package this binary so it can be used on the phone or elsewhere?
* Could you add some unit tests to ensure the code does what it's supposed to? Meaning, import creates messages as you expect.

**Comment by delijati** [comment]
Finally found some time to apply the change requests

**Comment by delijati** [comment]
ahh c.. did it in the old branch :/

File: README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+- Convenient manager to access the entries `tools/reader`
+- Import android exported voice and text events `tools/backup` [1]
+
+[1] Using https://github.com/handschuh/Slight-backup on the android side to
+export.
+
+### Backup
+
+Copy & backup first your existing `history.sqlite` from your phone to your desktop. We
+are here assuming `account0` that is the first sim slot on your phone.
```
Comment (UniversalSuperBox) [inline README.md]: This section is about restoring a pre-made backup from Android, right?

Maybe it's better to call it "Restore Android voice calls and text history"?

Also, you might want to add a few pointers on using the Slight-backup tool on Android, including how to get its backup files onto your computer.

**Comment by lduboeuf** [comment]
@delijati any news ? as this feature seems to be requested often.

File: README.md (hunk 4 above)
Comment (delijati) [reply README.md]: Done

**Comment by Flohack74** [comment]
I think we can clean up the branch ourselves and fix it.

**Comment by delijati** [comment]
Again done. This will be my last contribution to this issue. I hope it can be finally merged.

**Comment by Flohack74** [comment]
Yeah we are fixing now CI stuff, and then I can merge it probably...

**Comment by Flohack74** [comment]
One small question, did you test if the backup tool is shipped with the debian package correctly?

**Comment by delijati** [comment]
> One small question, did you test if the backup tool is shipped with the debian package correctly?

Sorry no not really.

**Comment by Flohack74** [comment]
I will trigger a rebuild and see whats happening.

**Comment by Flohack74** [comment]
Or, better yet, could I ask you to rebase all on the latest xenial, which i snow green in the CI? Then we are sure it builds the correct packages.

**Comment by delijati** [comment]
@Flohack74 it is already up to date
```
history-service on  xenial-android-import [?] via 🐍 v3.6.7 (py3) on ☁️  eu-central-1 took 11s 
❯ git merge xenial                                                                                                                                                                                       :(
Already up to date.

history-service on  xenial-android-import [?] via 🐍 v3.6.7 (py3) on ☁️  eu-central-1 
❯ git push        
Everything up-to-date

```

**Comment by Flohack74** [comment]
Hmm I just see that you might need to add the new subdirectory here: https://github.com/delijati/history-service/blob/xenial-android-import/tools/CMakeLists.txt

**Comment by Flohack74** [comment]
@delijati I am sorry to bother you again, but we have now fixed the unstable tests hopefully, can I ask you to rebase one last time to master ? =) It should be mergeable then (if the addin is correctly packaged in the deb file)

**Comment by delijati** [comment]
> @delijati I am sorry to bother you again, but we have now fixed the unstable tests hopefully, can I ask you to rebase one last time to master ? =) It should be mergeable then (if the addin is correctly packaged in the deb file)

lgtm now :)

**Comment by UniversalSuperBox** [comment]
We've moved this repository to GitLab, so we won't be merging any more pull requests from GitHub. If you (or anyone reading this) believe these changes are still relevant, please re-propose them at https://gitlab.com/ubports/core/history-service.

Thank you!

**Pull request closed without merging**